project/tests/com/google/daggerquery/executor/models/QueryTest.java

```diff
@@ -111,6 +111,28 @@ public void testExecutingDepsQuery_WithNullBindingGraph_ThrowsNullPointerExcepti
     query.execute(null);
   }
 
+  @Test
+  public void testExecutingDepsQuery_WithOneTypoInNodeName_ThrowsMisspelledNodeNameException() {
+    try {
+      String[] parameters = {"com.google.CatsFactoryy"};
+      Query query = new Query("deps", parameters);
+
+      List<String> queryExecutionResult = query.execute(makeSimpleBindingGraph());
+      fail();
+    } catch (MisspelledNodeNameException e) {
+      // We do not care too much about the rest of the message, but it should contain a correct node name.
+      assertTrue(e.getMessage().contains("com.google.CatsFactory"));
+    }
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testExecutingDepsQuery_WithFourTyposInNodeName_ThrowsIllegalArgumentException() {
```

Comment:
nit: How about changing `Four` to `TooMany` to make it more descriptive?

Reply:
Good idea, done ✅